### PR TITLE
Feat/handle errors

### LINKED ISSUE #11: Handle User / Tickets errors
For now all errors return "unknown error", should have a specific error message for specific erros, like:

- User Sign UP: CPF / Email already exist
- User Sign In: Invalid username or password
.
.
.

### PR DESCRIPTION
**Pull request by VerasThiago**
Closes #11

### CODE REVIEW

File: shared/repository/postgresRepository/ticket.go [hunk 1]
```diff
@@ -5,62 +5,42 @@ import (
...
-	if record := p.db.Where("owner_id = ?", userID).Find(&ticketList); record.Error != nil {
-		if record.Error == gorm.ErrRecordNotFound {
-			return nil, errors.GenericError{
-				Code:    errors.STATUS_NOT_FOUND,
-				Err:     record.Error,
-				Message: "Ticket not found",
-			}
-		}
-		return nil, record.Error
+	if err := errors.HandleDataNotFoundError(p.db.Where("owner_id = ?", userID).Find(&ticketList).Error, TICKET_MODEL_NAME); err != nil {
```
Comment (limarodrigoo) [inline shared/repository/postgresRepository/ticket.go:14]: Do u think that error handling should be inside of a db's methods implementation?

File: shared/errors/helper.go [hunk 2]
```diff
@@ -15,6 +19,8 @@ const (
 	STATUS_INTERNAL_SERVER_ERROR = http.StatusInternalServerError
 	STATUS_UNAUTHORIZED          = http.StatusUnauthorized
 	STATUS_NOT_FOUND             = http.StatusNotFound
+
+	DUPLICATED_KEY = "23505"
```
Comment (diogoftp) [inline shared/errors/helper.go:23]: How about a more meaningful name?

File: shared/repository/postgresRepository/ticket.go (hunk 1 above)
Comment (VerasThiago) [reply shared/repository/postgresRepository/ticket.go:14]: On that case, it's an error package with all errors related, what we can do is separate by files

File: shared/errors/helper.go [hunk 3]
```diff
@@ -50,3 +56,47 @@ func CreateGenericErrorFromValidateError(errors []string) GenericError {
...
+func HandleDuplicateError(err error) error {
+	if err == nil {
+		return err
+	}
+
+	if IsDuplicatedKeyError(err) {
+		return GenericError{
+			Code:    STATUS_BAD_REQUEST,
+			Err:     err,
+			Message: "Data already used",
```
Comment (diogoftp) [inline shared/errors/helper.go:81]: How about "Data is already being used"

**Review by diogoftp — APPROVED** [review]

**Review by diogoftp — APPROVED** [review]

**Pull request merged**